src/gt.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Returns true if the first argument is greater than the second; false otherwise.
+ *
+ * @example
+ * ```ts
+ * gt(5, 1) // expected true
+ * gt(1, 5) // expected false
+ * gt("a", "b") // expected false
+ * gt("b", "a") // expected true
+ *
+ * filter(gt(5), [1, 2, 4, 5, 8, 9]) // Iterable<[8, 9]>
+ * filter(gt(5), [1, 2, 3, 4, 5]) // Iterable<[]>
+ * filter(gt("b"), ["a", "b", "c"]) // Iterable<["c"]>
+ * filter(gt("b"), ["a", "b"]) // Itreable<[]>
+ * ```
+ */
+
+function gt(a: string | number | Date): (b: string | number | Date) => boolean;
+function gt(a: string, b: string): boolean;
+function gt(a: number, b: number): boolean;
+function gt(a: Date, b: Date): boolean;
+
+function gt(
+  a: string | number | Date,
+  _b?: string | number | Date,
+): ((b: string | number | Date) => boolean) | boolean {
+  if (_b !== undefined) {
+    return a > _b;
+  }
+
+  return (b: string | number | Date): boolean => {
+    if (typeof a !== typeof b) {
+      throw new TypeError(
+        "The values you want to compare must be of the same type",
+      );
+    }
+    return b > a;
+  };
+}
```

Comment:
'gt' of 'lodash' does not provide 'curry'.
`ramda`... is I don't know

The reason why the order has changed is because of `curry`.

The test code is as follows:
```ts
it("given array then should return greater values", function () {
      const result = pipe([4, 5, 6], filter(gt(5)), toArray);
      expect(result).toEqual([6]);
});
```

it is faild at `gt(a, _b)` but `gt(_b, a)` is success

```ts
function gt(a: any, b?: any): ((b: any) => boolean) | boolean {
  if (b === undefined) {
    return (_b: any) => gt(_b, a);
  }

  if (a.constructor !== b.constructor) {
    throw new TypeError(
      "The values you want to compare must be of the same type",
    );
  }

  return a > b;
}
```

Reply:
### A
```ts
pipe([4, 5, 6], filter(gt(5)), toArray); // [6]
```

### B
```ts
pipe([4, 5, 6], filter(lt(5)), toArray); // [6]
```

I agree that A looks more natural than B. However, many users will think that `gt(5)` is a curryed form of `gt(5, x)`, so it is easy to predict that it will evaluate to `5 > x`.

Reply:
It's vague, but I understand.

I will work to change test code